File: TASK-03/go/subtask4.go
package main

import (
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
    data, _ := ioutil.ReadFile("input.txt")
    n, _ := strconv.Atoi(strings.TrimSpace(string(data)))

    var output strings.Builder
    for i := 0; i <= n/2; i++ {
        output.WriteString(strings.Repeat(" ", n/2-i) + strings.Repeat("*", 2*i+1) + "\n")
    }
    for i := n/2 - 1; i >= 0; i-- {
        output.WriteString(strings.Repeat(" ", n/2-i) + strings.Repeat("*", 2*i+1) + "\n")
    }
    ioutil.WriteFile("output.txt", []byte(output.String()), 0644)
}

File: TASK-03/haskell/subtask4.hs
import System.IO

main :: IO ()
main = do
    contents <- readFile "input.txt"
    let n = read contents :: Int

    let upper = [0 .. div n 2]
        lower = reverse [0 .. div n 2 - 1]
        diamond = map (printLine n) upper ++ map (printLine n) lower

    writeFile "output.txt" (unlines diamond)

printLine :: Int -> Int -> String
printLine n i = replicate (div n 2 - i) " " ++ replicate (2 * i + 1) "*"
